CDTDatastore/touchdb/TDPuller.m

```diff
@@ -303,42 +354,55 @@ - (void)processInbox:(TD_RevisionList*)inbox
 // Add a revision to the appropriate queue of revs to individually GET
 - (void)queueRemoteRevision:(TD_Revision*)rev
 {
-    if (rev.deleted) {
-        if (!_deletedRevsToPull) _deletedRevsToPull = [[NSMutableArray alloc] initWithCapacity:100];
-
-        [_deletedRevsToPull addObject:rev];
+    if (_bulkGetSupported) {
+        [_bulkGetRevs addObject:rev];
     } else {
-        if (!_revsToPull) _revsToPull = [[NSMutableArray alloc] initWithCapacity:100];
-
-        [_revsToPull addObject:rev];
+        if (rev.deleted) {
+            [_deletedRevsToPull addObject:rev];
+        } else {
+            [_revsToPull addObject:rev];
+        }
     }
 }
 
 // Start up some HTTP GETs, within our limit on the maximum simultaneous number
 - (void)pullRemoteRevisions
 {
     while (_db && _httpConnectionCount < kMaxOpenHTTPConnections) {
-        NSUInteger nBulk = MIN(_bulkRevsToPull.count, kMaxRevsToGetInBulk);
-        if (nBulk == 1) {
-            // Rather than pulling a single revision in 'bulk', just pull it normally:
-            [self queueRemoteRevision:_bulkRevsToPull[0]];
-            [_bulkRevsToPull removeObjectAtIndex:0];
-            nBulk = 0;
-        }
-        if (nBulk > 0) {
-            // Prefer to pull bulk revisions:
+        
+        // _bulkGetRevs count != 0 means _bulk_get endpoint supported
+        if ([_bulkGetRevs count] !=0) {
```

Comment:
Can't we just check `_bulkGetSupported`? We could then do away with the comment too.